### PR TITLE
Googleapi fix

### PR DESCRIPTION
**Pull request by danner26**
Fixed the API for Google and added handling. Also added another CLI switch for manual key input
#27

### CODE REVIEW

**Comment by danner26** [comment]
I think i'm also going to add in a fallback switch/option. The way I see it, if the Apple lookup fails, or the Google lookup fails, you can specify the switch to try and fall back to the other service. If both fails, then report so, otherwise return the proper data.

Just an idea.. not too sure how practical this is

**Comment by clburlison** [comment]
@danner26 Thank you for working on this. I can see people liking the fallback option. I'll review what you have end of day. On first glance everything looks pretty.

**Comment by danner26** [comment]
I changed it around a little so there is logging and it doesn't depend on the requests module (AKA no extra module installs)
Otherwise everything appears to be working

**Comment by danner26** [comment]
Tested with `sudo ./pinpoint -f -l google -a google -k {MY_API_KEY} -vv` for reference

**Comment by danner26** [comment]
I didn't rewrite the reverse lookup yet, working on that now

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,7 @@ tmp/
 .pyc
```
Comment (clburlison) [inline .gitignore]: So this was a mistake on my part. Could you please change this line to be `*.pyc` and remove lines 7-9 that you added?

File: pkgroot/Library/Application Support/pinpoint/bin/pinpoint [hunk 2]
```diff
@@ -327,24 +328,40 @@ def lookup(service, time=None):
...
+          "considerIp": "false",
+          "wifiAccessPoints": []
+        }
         count = 0
         for i, val in enumerate(scan):
-            api_url += ('&wifi=mac:{0}&ssid:{1}&ss:{2}'.format(
-                         val.get('BSSID'),
-                         val.get('SSID_STR'),
-                         val.get('RSSI')))
+            request_data['wifiAccessPoints'].append({'macAddress':val.get('BSSID'), 'signalStrength':val.get('RSSI'), 'signalToNoiseRatio':0})
```
Comment (clburlison) [inline pkgroot/Library/Application Support/pinpoint/bin/pinpoint]: This line is too long and fails linting. Please change to:
```python
            request_data['wifiAccessPoints'].append({
                'macAddress': val.get('BSSID'),
                'signalStrength': val.get('RSSI'),
                'signalToNoiseRatio': 0}
```

File: pkgroot/Library/Application Support/pinpoint/bin/pinpoint [hunk 3]
```diff
@@ -327,24 +328,40 @@ def lookup(service, time=None):
...
             count += 1
             if count == 5:
                 break
-
-        data = download_file(api_url)
-        if data is not False:
-            obj = json.loads(data)
+        dump = json.dumps(request_data).encode("utf8")
+        try:
+            request = Request(api_url, dump, {'Content-type': 'application/json'})
```
Comment (clburlison) [inline pkgroot/Library/Application Support/pinpoint/bin/pinpoint]: Line is too long and fails linting. Please change to:
```python
            request = Request(api_url, dump,
                              {'Content-type': 'application/json'}
```

**Comment by clburlison** [comment]
I've updated the docs. https://github.com/clburlison/pinpoint/wiki/Preferences/_compare/9ca694a037a3991816b967ff68a415d485045092...9ee784f3fd92a324788b042bc58561639c9588ac

**Comment by danner26** [comment]
Changes have been implemented. Should be good now.

I will create a new pull request once I finish the fallback option.

**Pull request merged**